lib/membrane_rtc_engine/engine.ex

```diff
@@ -497,18 +483,40 @@ defmodule Membrane.RTC.Engine do
         &MapSet.union(&1, new_waiting_for_linking)
       )
 
+    new_endpoint_subscriptions =
+      Map.new(new_outbound_tracks, fn {track_id, format} -> {track_id, format} end)
+
+    subscriptions =
+      Map.update(
+        state.subscriptions,
+        endpoint_id,
+        new_endpoint_subscriptions,
+        &Map.merge(&1, new_endpoint_subscriptions)
+      )
+
+    state = %{state | subscriptions: subscriptions}
     {{:ok, [spec: %ParentSpec{links: new_links}]}, state}
   end
 
   @impl true
-  def handle_notification({:new_tracks, tracks}, {:endpoint, endpoint_id}, ctx, state) do
+  def handle_notification(
+        {:publish, {:new_tracks, tracks}},
+        {:endpoint, endpoint_id},
+        ctx,
+        state
+      ) do
     id_to_track = Map.new(tracks, &{&1.id, &1})
 
     state =
       update_in(state, [:endpoints, endpoint_id, :inbound_tracks], &Map.merge(&1, id_to_track))
 
     tracks_msgs =
-      update_track_messages(state.endpoints, ctx, {:add_tracks, tracks}, {:endpoint, endpoint_id})
+      update_track_messages(
```

Comment:
I think we should rename this function. We got notification to publish something so we could name this function `publish`.